src/models/Popup.js

```diff
@@ -417,12 +473,27 @@ export default class Popup extends Base {
       el.classList.add('cc-invisible')
     }
 
+    const currentStatuses = this.getStatusesMap();
+
+    const openConsentsElem = document.getElementById(this.options.consentSettingsElemId);
+    if (openConsentsElem) {
+      openConsentsElem.onclick = () => this.openCustomizeSettings();
+    }
+
     el.addEventListener('click', event => this.handleButtonClick( event ) )
-    el.querySelectorAll( '.cc-btn [type="checkbox"]' ).forEach( checkbox => {
-      checkbox.addEventListener( 'change', () => {
-        this.userCategories[ checkbox.name ] = checkbox.checked ? 'ALLOW' : 'DENY'
-      })
-      checkbox.addEventListener( 'click', event => (event.stopPropagation()) )
+    el.querySelectorAll('.cc-category label [type="checkbox"]').forEach(checkbox => {
+      if (checkbox.name === 'ESSENTIAL') {
+        checkbox.checked = true;
+        checkbox.disabled = true;
+      } else {
+        this.userCategories[checkbox.name] = currentStatuses[checkbox.name] ? 'ALLOW' : 'DENY';
+        checkbox.checked = currentStatuses[checkbox.name] === 'ALLOW';
+        checkbox.addEventListener('change', () => {
+          this.userCategories[checkbox.name] = checkbox.checked ? 'ALLOW' : 'DENY'
+        })
+      }
+
+      checkbox.addEventListener('click', event => (event.stopPropagation()))
```

Comment:
As the callback is a statement, not an expression, please wrap it within braces (no implicit return):

```suggestion
      checkbox.addEventListener('click', event => { event.stopPropagation(); })
```

Reply:
What about just removing the parenthesis and not introducing braces?